### PR TITLE
Documentation updates

### PR DESCRIPTION
**Pull request by HarminderSingh1**
Addine a few more documentation updates - added FAQs section and added sequence diagrams for initializing the widget and starting the call.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -269,58 +271,57 @@ onDialNumber(data) {
...
-```
+### Initializing the call widget
 
-#### Navigate to a contacts/company page and launch call
+The following messages are exchanged when a call widget is instantiated.
+![Image description](./docs/images/InitializeCallWidgetIFrame.png)
+Once the widget iFrame is created, Hubspot send the SYNC message to the widget after iFrame is loaded and repetedly send this message until it receives the SYNC_ACK response from the widget. If the SYNC_ACK response isn't received within 30 seconds, the widget is marked as failed. Note that sending SYNC/SYNC_ACK messages are handled by the framework. Once the widget and host page are synchronized, the frameworks triggers the ready event.
 
-The calling extension demo widget should load inside an iFrame.
+The call widget should waits for the ready event from the framework and send the initialized event to HubSpot. At this point, the messages can be exchanged between the call widget and HubSpot.
```
Comment (anthony-nardi) [inline README.md]: super nit: `waits` -> `wait`

File: README.md [hunk 2]
```diff
@@ -269,58 +271,57 @@ onDialNumber(data) {
...
+Here is description of events:
 
-1. [HubSpot] Sends a SYNC message
-2. [SoftPhone] Sends SYNC_ACK message
-3. [SoftPhone] Sends INITIALIZED message with login state and optionally widget size
+1. **Dial number** - HubSpot sends the dial number event.
+2. **Outbound call started** - Widget notifies HubSpot when the call is started.
+3. **Create engagement** - HubSpot creates an engagement with minimum information if requested by the widget.
+4. **Engagement created** - HubSpot creates an engagement.
+5. **Engagement created** - HubSpot sends the engagementId to the widget.
```
Comment (anthony-nardi) [inline README.md]: Should there be 2 `**Engagement created**` here?

File: README.md [hunk 3]
```diff
@@ -269,58 +271,57 @@ onDialNumber(data) {
...
 
-1. [HubSpot] Sends DIAL_NUMBER message with phone_number to dial
-2. [SoftPhone] Sends the OUTGOING_CALL_STARTED message with the phone number that is dialed
-3. [SoftPhone] Sends the CALL_ANSWERED message
-4. [SoftPhone] Sends the CALL_ENDED message
+<details>
+ <summary>When an engagement should be created vs updated.</summary>
+ <p>
+    A user can initiate a call from inside the HubSpot UI and outside of the HubSpot UI (e.g. mobile app/redirected number/etc.)  If a call is initiated from within HubSpot UI, HubSpot will create a call engagement and send the engagement to the call widget.  Once the call finishes, the call widget can update this engagement with additional call details.  If a call is initiated outside of HubSpot UI, the widget should create the call engagement. 
+</p>
```
Comment (anthony-nardi) [inline README.md]: I'm curious about this. Above we mention that `callInfo` has a `createEngagement` property. How does that relate to "When an engagement should be created vs updated."?

File: README.md (hunk 2 above)
Comment (HarminderSingh1) [reply README.md]: I will revisit the names.

File: README.md (hunk 3 above)
Comment (HarminderSingh1) [reply README.md]: w/o the HubSpot UI those properties play no role - in that case the widget should create an engagement.  createEngagement is flag for our FE to create an engagement.

**Pull request merged**